### PR TITLE
Fix: Pointer reference error during destinationrule merge

### PR DESCRIPTION
**Pull request by ydh926**
Please provide a description for what this PR is for.

### CODE REVIEW

**Comment by howardjohn** [comment]
What is the issue?

/retest

**Comment by ydh926** [comment]
https://github.com/istio/istio/issues/22147 @howardjohn

**Comment by ydh926** [comment]
we should copy destinationrule instead of use the pointer of it. 
For the result of merging the same namespace and the result of merging cross namespaces will point to the same object，if we use pointer. @howardjohn

**Comment by hzxuzhonghu** [comment]
What's the bad effect of this as we would not mutate them?

**Comment by ydh926** [comment]
please see this issue https://github.com/istio/istio/issues/22147. @hzxuzhonghu

File: pilot/pkg/model/push_context_test.go [hunk 1]
```diff
@@ -21,6 +21,8 @@ import (
 	"testing"
 	"time"
 
+	"istio.io/istio/pkg/config/visibility"
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/push_context_test.go]: make this grouped with istio.io/isto

Comment (ydh926) [reply pilot/pkg/model/push_context_test.go]: Done

**Pull request merged**